src/custom-vocabularies-client.ts

```diff
@@ -0,0 +1,57 @@
+import * as FormData from 'form-data';
+import * as fs from 'fs';
+import { Readable } from 'stream';
+
+import { ApiRequestHandler } from './api-request-handler';
+import { CustomVocabulary } from './models/CustomVocabulary';
+import { CustomVocabularyOptions } from './models/CustomVocabularyOptions';
+import { RevAiApiCustomVocabulary } from './models/RevAiApiCustomVocabulary';
+
+/**
+* Client which handles communication with custom vocabularies
+* Rev.ai api
+*/
+export class RevAiCustomVocabulariesClient {
+    apiHandler: ApiRequestHandler;
+
+    /**
+    * @param accessToken Access token used to authenticate API requests
+    * @param version (optional) version of the API to be used
+    */
+    constructor(accessToken: string, version: string = 'v1') {
+        this.apiHandler = new ApiRequestHandler(`https://api.rev.ai/revspeech/${version}/vocabularies`, accessToken);
+    }
+
+    public async submitCustomVocabularies(
+        customVocabularies: CustomVocabulary[],
+        callbackUrl: string = undefined,
+        metadata: string = undefined
+    ): Promise<RevAiApiCustomVocabulary> {
+        if (!customVocabularies) {
+            throw Error('Custom Vocabularies not provided');
+        }
+
+        let options: CustomVocabularyOptions = {custom_vocabularies: customVocabularies};
+        if (callbackUrl) {
```

Comment:
I wonder if there is a 3rd party assertion library that we can use to validate params.

Reply:
I did a quick search. There don't appear to be any great libraries for this. I don't think it would be hard to write our own simple argument asserter
Something like: `argument.is(arg, func<argtype, bool>, failureMessage)` which does something like this when called
```
assert = require('assert')
function is(arg, testFunc, failureMessage) {
    try:
        assert.equal(testFunc(arg), true)
   catch:
       throw ArgumentError(failureMessage)
}
```

Reply:
Validate to what degree? I'm assuming that we don't need to validate much on the SDK side besides the existence of the parameter. All other parameter problems should be reported and returned by the API, no?

Reply:
Kostas and I discussed. We will resolve this comment.